### ISSUE TITLE
[BUG] ShapeletTransformClassifier doesn't work with y as pd.Series

### ISSUE THREAD

**Issue by mloning** [alan-turing-institute/sktime#321]
**Describe the bug**
 ShapeletTransformClassifier doesn't work with y as pd.Series

**To Reproduce**
This breaks: 
```python
from sktime.classification.shapelet_based import ShapeletTransformClassifier
from sktime.datasets import load_arrow_head
from sklearn.model_selection import train_test_split

X, y = load_arrow_head(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y)

c = ShapeletTransformClassifier()
c.fit(X_train, y_train)
```

**Expected behavior**
Works with pd.Series or np.array

**Additional context**
This works: 

```python
from sktime.classification.shapelet_based import ShapeletTransformClassifier
from sktime.datasets import load_arrow_head
from sklearn.model_selection import train_test_split

X, y = load_arrow_head(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y)

c = ShapeletTransformClassifier()
c.fit(X_train, y_train.to_numpy())
```

**Comment by mloning** [comment]
This may be fixed by #287

**Comment by TonyBagnall** [comment]
sorry, forgot about this, I'm in Java land. I suggest we just internally check if its passed y as a pd.Series and if so, pull it out as an array? Should be a quick fix?

**Comment by TonyBagnall** [comment]
@ABostrom want to do this?

**Comment by ABostrom** [comment]
I'll do it now. should take me 10mins. (he says confidently)

**Comment by mloning** [comment]
Adding `y_train = y_train.to_numpy()` inside fit should probably do it

**Comment by mloning** [comment]
Closed by #329
